ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/HttpResponseImpl.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client.impl;
+
+import io.confluent.ksql.api.client.Client.HttpResponse;
+import io.confluent.ksql.api.client.exception.KsqlClientException;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.JsonObject;
+import java.util.Collections;
+import java.util.Map;
+
+public class HttpResponseImpl implements HttpResponse {
+
+  private final int status;
+  private final byte[] body;
+
+  public HttpResponseImpl(final int status, final byte[] body) {
+    this.status = status;
+    this.body = body;
```

Comment:
Null check needed since arrays can be null

Reply:
response body can be empty, right? it should be fine to leave this as it is. we handle the null case when converting it to a map in `bodyAsMap()`